### ISSUE TITLE
Do you have working suspend?

### ISSUE THREAD

**Issue by baybal** [BNieuwenhuizen/zenbook-s16#3]

**Comment by ThatOneCalculator** [comment]
Suspend sort of works for me. It does actually go into suspension, but they keyboard lights stay on and the screen is frozen for about 10 seconds after waking up.

**Comment by BNieuwenhuizen** [comment]
Was the same situation as @ThatOneCalculator for me. However, I have an upstream bug at https://gitlab.freedesktop.org/drm/amd/-/issues/3545 , and the new patches linked there seem to do something (the keyboard backlight now turns off in suspend at least)

**Comment by ThatOneCalculator** [comment]
Rebuilding my kernel now! If it works, I'll include the patch in my AUR package :)

**Comment by ThatOneCalculator** [comment]
Update -- tested, works, and pushed! I still get the freezing for 10 seconds, but the keyboard backlight does in fact turn off.

**Comment by baybal** [comment]
> Suspend sort of works for me. It does actually go into suspension, but they keyboard lights stay on and the screen is frozen for about 10 seconds after waking up.

For me, it goes to suspend, and never comes back properly (the side power led turns on, screen black)

**Comment by ThatOneCalculator** [comment]
@baybal What distro/kernel/firmware version are you on?

**Comment by baybal** [comment]
Gentoo, vanilla 6.11-rc4 from git

**Comment by ThatOneCalculator** [comment]
I'd recommend trying to manually apply the three patches in https://aur.archlinux.org/packages/linux-mainline-um5606

- https://aur.archlinux.org/cgit/aur.git/tree/zsc_control.patch?h=linux-mainline-um5606
- https://aur.archlinux.org/cgit/aur.git/tree/npu.patch?h=linux-mainline-um5606
- https://patchwork.kernel.org/project/bluetooth/patch/20240804195319.3499920-1-bas@basnieuwenhuizen.nl/mbox/

**Comment by hworld** [comment]
Curious if these patches fixe the keyboard brightness control not working as well, since it turns off the backlight during suspend. Or is that still not working yet?

I did some energy consumption tests and it seems like the backlight brings it up a couple watts for me (6.5w vs 4w idle), which is pretty nuts.

**Comment by ThatOneCalculator** [comment]
> Or is that still not working yet?

Not yet. The patch specifically tells it to go into s2idle, and the firmware knows that the keyboard backlight should be off during that (as far as I know!)

**Comment by baybal** [comment]
Somehow, I got a first suspend to ram working without patches. We will see if it will go to suspend a second time. No PSR disable, no other special measures

**Comment by baybal** [comment]
Unfortunately it only lasted for 1 reboot. It seems there is some system setting getting either touched by the firmware, or Linux PM code which makes the difference

**Comment by baybal** [comment]
> > Or is that still not working yet?
> 
> Not yet. The patch specifically tells it to go into s2idle, and the firmware knows that the keyboard backlight should be off during that (as far as I know!)

Can you try suspend to ram instead of s2idle?

**Comment by BNieuwenhuizen** [comment]
I don't think this laptop supports S3 at all, BIOS-wise. (from `/sys/power/mem_sleep`)

**Comment by ThatOneCalculator** [comment]
I'd highly recommend just using the submitted patch for working s2idle.

**Comment by baybal** [comment]
> I don't think this laptop supports S3 at all, BIOS-wise. (from `/sys/power/mem_sleep`)

I did not know it is not even supported

**Comment by ThatOneCalculator** [comment]
This issue should be closed now due to https://wiki.archlinux.org/title/ASUS_Zenbook_UM5606#Power_management
